File: PROGRAMMERS/LEVEL_02/Q12985.cpp
//https://programmers.co.kr/learn/courses/30/lessons/12985
//예상 대진표
#include <iostream>

using namespace std;

int solution(int n, int a, int b)
{
    int cnt = 0;
    // a = b가 되면 종료
    while (a != b)
    {
        // a와 b가 같지 않으면 대진
        ++cnt;

        // a 와 b의 다음 대진표 번호 구하기
        a = (a + 1) / 2;
        b = (b + 1) / 2;
    }

    return cnt;
}

int main()
{

    int n = 8; int a = 4; int b = 7;

    cout << solution(n, a, b);

    return 0;
}

File: PROGRAMMERS/LEVEL_02/Q12985.java
// https://programmers.co.kr/learn/courses/30/lessons/12985?language=java
// 예상 대진표
package PROGRAMMERS.LEVEL_02;

public class Q12985 {
    public static void main(String[] args) {
        /*
        N	A	B	answer
        8	4	7	3

         */

        int n = 8;
        int a = 4;
        int b = 7;
        System.out.println(solution(n, a, b));
    }

    public static int solution(int n, int a, int b) {

        return Integer.toBinaryString((a - 1) ^ (b - 1)).length();
    }
}
